logs/config_score_83.4000.txt
Description: CIFAR classification with CNN using Pytorch
DATA_FOLDER: ../data/
MODEL_FOLDER: ../models/
LOGS_FOLDER: ../logs/
best_model_path: ../models/best_model-177586.pth
description: CIFAR classification with CNN using Pytorch
seed: 42
device: cuda
num_workers: 20
num_classes: 10
batch_size: 1024
epochs: 100
lr: 0.0025
weight_decay: 1e-06
momentum: 0.9
image_size: 28
aug_prob: 0.5
angle_range: 15
translate_range: 0.1
scale_range: (0.9, 1.1)
shear_range: 10
=================================================================
Layer (type:depth-idx)                   Param #
=================================================================
CifarNet                                 --
├─Sequential: 1-1                        --
│    └─ConvBlock: 2-1                    --
│    │    └─Conv2d: 3-1                  896
│    │    └─BatchNorm2d: 3-2             64
│    │    └─ReLU: 3-3                    --
│    │    └─MaxPool2d: 3-4               --
│    └─ConvBlock: 2-2                    --
│    │    └─Conv2d: 3-5                  18,496
│    │    └─BatchNorm2d: 3-6             128
│    │    └─ReLU: 3-7                    --
│    │    └─MaxPool2d: 3-8               --
│    └─ConvBlock: 2-3                    --
│    │    └─Conv2d: 3-9                  73,856
│    │    └─BatchNorm2d: 3-10            256
│    │    └─ReLU: 3-11                   --
│    │    └─MaxPool2d: 3-12              --
│    └─ConvBlock: 2-4                    --
│    │    └─Conv2d: 3-13                 295,168
│    │    └─BatchNorm2d: 3-14            512
│    │    └─ReLU: 3-15                   --
│    │    └─MaxPool2d: 3-16              --
│    └─ConvBlock: 2-5                    --
│    │    └─Conv2d: 3-17                 1,180,160
│    │    └─BatchNorm2d: 3-18            1,024
│    │    └─ReLU: 3-19                   --
│    │    └─MaxPool2d: 3-20              --
├─Linear: 1-2                            262,656
├─Linear: 1-3                            131,328
├─Linear: 1-4                            2,570
├─ReLU: 1-5                              --
=================================================================
Total params: 1,967,114
Trainable params: 1,967,114
Non-trainable params: 0
=================================================================


logs/config_score_84.3600.txt
Description: CIFAR classification with CNN using Pytorch
DATA_FOLDER: ../data/
MODEL_FOLDER: ../models/
LOGS_FOLDER: ../logs/
best_model_path: ../models/best_model-151695.pth
description: CIFAR classification with CNN using Pytorch
seed: 42
device: cuda
num_workers: 20
num_classes: 10
batch_size: 1024
epochs: 100
lr: 0.0025
weight_decay: 1e-06
momentum: 0.9
image_size: 28
aug_prob: 0.5
angle_range: 15
translate_range: 0.1
scale_range: (0.9, 1.1)
shear_range: 10
=================================================================
Layer (type:depth-idx)                   Param #
=================================================================
CifarNet                                 --
├─Sequential: 1-1                        --
│    └─ConvBlock: 2-1                    --
│    │    └─Conv2d: 3-1                  896
│    │    └─BatchNorm2d: 3-2             64
│    │    └─ReLU: 3-3                    --
│    │    └─MaxPool2d: 3-4               --
│    └─ConvBlock: 2-2                    --
│    │    └─Conv2d: 3-5                  18,496
│    │    └─BatchNorm2d: 3-6             128
│    │    └─ReLU: 3-7                    --
│    │    └─MaxPool2d: 3-8               --
│    └─ConvBlock: 2-3                    --
│    │    └─Conv2d: 3-9                  73,856
│    │    └─BatchNorm2d: 3-10            256
│    │    └─ReLU: 3-11                   --
│    │    └─MaxPool2d: 3-12              --
│    └─ConvBlock: 2-4                    --
│    │    └─Conv2d: 3-13                 295,168
│    │    └─BatchNorm2d: 3-14            512
│    │    └─ReLU: 3-15                   --
│    │    └─MaxPool2d: 3-16              --
│    └─ConvBlock: 2-5                    --
│    │    └─Conv2d: 3-17                 1,180,160
│    │    └─BatchNorm2d: 3-18            1,024
│    │    └─ReLU: 3-19                   --
│    │    └─MaxPool2d: 3-20              --
├─Linear: 1-2                            262,656
├─Linear: 1-3                            131,328
├─Linear: 1-4                            2,570
├─ReLU: 1-5                              --
=================================================================
Total params: 1,967,114
Trainable params: 1,967,114
Non-trainable params: 0
=================================================================


app/src/config/config.py
"""Configuration file.

Configuration of project variables that we want to have available
everywhere and considered configuration.
"""
import os
from dataclasses import dataclass
import multiprocessing

@dataclass 
class Configuration:
    """Configuration class for the project."""
    # =============== Paths =============== 
    DATA_FOLDER: str = "../data/"
    MODEL_FOLDER: str = "../models/"
    LOGS_FOLDER: str = "../logs/"
    PID = None

    best_model_path: str = os.path.join(MODEL_FOLDER, "best_model.pth")

    # =============== Variabless =============== 
    description: str = "MNIST classification with MLP using Pytorch"
    seed: int = 42
    device: str = "cuda"  # "cpu" or "cuda"
    num_workers: int = multiprocessing.cpu_count()-4
    num_classes: int = 10
    batch_size: int = 128
    epochs: int = 300
    lr: float = 0.1
    weight_decay: float = 5e-4
    momentum: float = 0.9
    dropout_rate: float = 0.4
    label_smoothing: float = 0.1
    aug_prob: float = 0.8

    # =============== Data aug parameters =============== 
    image_size: int = 28
    aug_prob: float = 0.5
    angle_range: int = 15  # degres
    translate_range: float = 0.1  # fraction of img size
    scale_range: tuple = (0.9, 1.1)
    shear_range: int = 10  # degres

    def __post_init__(self):
        self.PID = os.getpid()
        self.best_model_path = os.path.join(self.MODEL_FOLDER, f"best_model-{self.PID}.pth")
